cmd/kaspawallet/daemon/server/sync.go

```diff
@@ -274,3 +283,26 @@ func (s *server) refreshUTXOs() error {
 func (s *server) isSynced() bool {
 	return s.nextSyncStartIndex > s.keysFile.LastUsedInternalIndex() && s.nextSyncStartIndex > s.keysFile.LastUsedExternalIndex()
 }
+
+func (s *server) updateSyncingProgressLog(currProcessedAddresses, currMaxUsedAddresses uint32) {
+	if currMaxUsedAddresses > maxUsedAddressesForLog {
+		maxUsedAddressesForLog = currMaxUsedAddresses
```

Comment:
Please don't use global variables for state, instead, store the relevant data inside `server`. This makes testing easier.